tests/statistics_test.py

```diff
@@ -0,0 +1,243 @@
+
+        map_name = random_string() + ",t=es\\t"
+
+        near_cache_config = NearCacheConfig(map_name)
+        config.near_cache_configs[map_name] = near_cache_config
+
+        client = HazelcastClient(config)
+        test_map = client.get_map(map_name).blocking()
+
+        time.sleep(2 * self.STATS_PERIOD)
+        response = self._wait_for_statistics_collection(2 * self.STATS_PERIOD)
+
+        result = response.result.decode("utf-8")
+        unescaped_result = self._unescape_special_chars(result)
+        self.assertEqual(-1, result.find(map_name))
+        self.assertNotEqual(-1, unescaped_result.find(map_name))
+        client.shutdown()
+
+    def test_near_cache_stats(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, True)
+        config.set_property(ClientProperties.STATISTICS_PERIOD_SECONDS.name, self.STATS_PERIOD)
+
+        map_name = random_string()
+
+        near_cache_config = NearCacheConfig(map_name)
+        config.near_cache_configs[map_name] = near_cache_config
+
+        client = HazelcastClient(config)
+        test_map = client.get_map(map_name).blocking()
+
+        time.sleep(2 * self.STATS_PERIOD)
+        response = self._wait_for_statistics_collection(2 * self.STATS_PERIOD)
+
+        result = response.result.decode("utf-8")
+        self.assertEqual(1, result.count("nc." + map_name + ".evictions=0"))
+        self.assertEqual(1, result.count("nc." + map_name + ".hits=0"))
+        self.assertEqual(1, result.count("nc." + map_name + ".misses=0"))
+        self.assertEqual(1, result.count("nc." + map_name + ".ownedEntryCount=0"))
+        self.assertEqual(1, result.count("nc." + map_name + ".expirations=0"))
+        self.assertEqual(1, result.count("nc." + map_name + ".invalidations=0"))
+        self.assertEqual(1, result.count("nc." + map_name + ".invalidationRequests=0"))
+
+        test_map.put(1, 2)  # invalidation request
+        test_map.get(1)     # cache miss
+        test_map.get(1)     # cache hit
+        test_map.put(1, 3)  # invalidation + invalidation request
+        test_map.get(1)     # cache miss
+
+        time.sleep(2 * self.STATS_PERIOD)
+        response = self._wait_for_statistics_collection(2 * self.STATS_PERIOD)
+
+        result = response.result.decode("utf-8")
+        self.assertEqual(1, result.count("nc." + map_name + ".evictions=0"))
+        self.assertEqual(1, result.count("nc." + map_name + ".hits=1"))
+        self.assertEqual(1, result.count("nc." + map_name + ".misses=2"))
+        self.assertEqual(1, result.count("nc." + map_name + ".ownedEntryCount=1"))
+        self.assertEqual(1, result.count("nc." + map_name + ".expirations=0"))
+        self.assertEqual(1, result.count("nc." + map_name + ".invalidations=1"))
+        self.assertEqual(1, result.count("nc." + map_name + ".invalidationRequests=2"))
+
+        client.shutdown()
+
+    def _get_client_stats_from_server(self):
```

Comment:
I know this is same for other clients too, but this can result in bugs in the future. If there are some leftover clients in the tests then getting the first client might not be correct. Nodejs client has updated this:

```
    var clientUuid = client.getClusterService().uuid;
    var script =
        'clients=instance_0.getClientService().getConnectedClients().toArray()\n' +
        'for(i=0;i<clients.length;i++) {\n' +
        '   if (clients[i].getUuid().equals("' + clientUuid + '")) {\n' +
        '       result=clients[i].getClientStatistics();\n' +
        '       break;' +
        '   }\n' +
        '}\n';
```

Doing this is a safe way instead of assuming that there wont be any leftover clients.

Reply:
I will change this